sdk/storage/azblob/service/client.go

```diff
@@ -120,11 +120,8 @@ func (s *Client) URL() string {
 	return s.generated().Endpoint()
 }
 
-// NewContainerClient creates a new ContainerClient object by concatenating containerName to the end of
-// Client's URL. The new ContainerClient uses the same request policy pipeline as the Client.
-// To change the pipeline, create the ContainerClient and then call its WithPipeline method passing in the
-// desired pipeline object. Or, call this package's NewContainerClient instead of calling this object's
-// NewContainerClient method.
+// NewContainerClient creates a new container.Client object by concatenating containerName to the end of
+// this Client's URL. The new container.Client uses the same request policy pipeline as the Client.
 func (s *Client) NewContainerClient(containerName string) *container.Client {
```

Comment:
We need to do the same for container names too for consistency. Look at tiny blob methods too. Some of those methods take both a container & blob name

Reply:
@seanmcc-msft can you comment on this?  From the docs, I thought there are more stringent requirements on container names.  That said, if .NET is doing this, we can follow suit.

Reply:
I don't think you can do this. It is a breaking change to alter this behavior. URL Endpoint must be encoded by the customer before passing them to our ctor. But, methods that take high-level abstractions (like container or blob) we'll encode them before putting them into a URL - I think we just have to do this.